### PR TITLE
HPCC-30892 Fileserver fetch scripts for built assets

### PR DESCRIPTION
**Pull request by Michael-Gardner**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: build_utils/fetch_assets.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    os.makedirs(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}", exist_ok=True)
+    os.chdir(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}")
+    download_asset(f"{repo_url}/download/community_{version}/ALL_HPCC_DOCS_{language}-{version}.zip")
+    download_asset(f"{repo_url}/download/community_{version}/ECL_Code_Files_{language}.zip")
+  chown_recursive(f"{data_dir_base}/CE-Candidate-{base_version}", local_uid, local_gid)
+
+
+  ## Fetch internal assets from JFrog
+  logging.info(f"Creating {data_dir_base}/LN-Candidate-{base_version}/bin/platform directory")
+  os.makedirs(f"{data_dir_base}/LN-Candidate-{base_version}/bin/platform", exist_ok=True)
```
Comment (cloLN) [inline build_utils/fetch_assets.py:237]: this should go to LN-Candidate-{base_version}/bin/platform-withplugins

**Review by cloLN — CHANGES_REQUESTED** [review]
please change the target location of internal platform builds to go to /bin/platform-withplugins instead.

**Review by cloLN — CHANGES_REQUESTED** [review]
We need the unzip function included for the documentation as well.

File: build_utils/fetch_assets.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  download_asset(f"{repo_url}/download/community_{version}/hpccsystems-eclide-community_{version}Windows-i386.exe")
+  # do CE docs
+  for language in documentation_languages:
+    logging.info(f"Creating {data_dir_base}/CE-Candidate-{base_version}/docs/{language} directory")
+    os.makedirs(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}", exist_ok=True)
+    os.chdir(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}")
+    download_asset(f"{repo_url}/download/community_{version}/ALL_HPCC_DOCS_{language}-{version}.zip")
+    download_asset(f"{repo_url}/download/community_{version}/ECL_Code_Files_{language}.zip")
+  chown_recursive(f"{data_dir_base}/CE-Candidate-{base_version}", local_uid, local_gid)
+
```
[the remark this answers is not in the document]
Comment (Michael-Gardner) [reply build_utils/fetch_assets.py:233]: Agreed. I'll get that functionality added

File: build_utils/fetch_assets.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  download_asset(f"{repo_url}/download/community_{version}/hpccsystems-eclide-community_{version}Windows-i386.exe")
+  # do CE docs
+  for language in documentation_languages:
+    logging.info(f"Creating {data_dir_base}/CE-Candidate-{base_version}/docs/{language} directory")
+    os.makedirs(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}", exist_ok=True)
+    os.chdir(f"{data_dir_base}/CE-Candidate-{base_version}/docs/{language}")
+    download_asset(f"{repo_url}/download/community_{version}/ALL_HPCC_DOCS_{language}-{version}.zip")
+    download_asset(f"{repo_url}/download/community_{version}/ECL_Code_Files_{language}.zip")
+    if os.path.exists(f"ALL_HPCC_DOCS_{language}-{version}.zip"):
+      with zipfile.ZipFile(f"ALL_HPCC_DOCS_{language}-{version}.zip", 'r') as zipped_docs:
```
Comment (Michael-Gardner) [inline build_utils/fetch_assets.py:234]: We now unzip the ALL_HPCC_DOCS file in the docs/language directory

Comment (cloLN) [reply build_utils/fetch_assets.py:234]: Looks good.

**Review by cloLN — COMMENTED** [review]
Looks good to merge.

**Review by cloLN — APPROVED** [review]
Looks good to merge.  Approving change.

**Comment by Michael-Gardner** [comment]
@ghalliday Ready for merging. Putting this in the build_utils folder. It's infrastructure code for the platforms build/publishing.

**Pull request merged**